components/formats-bsd/src/loci/formats/services/EXIFServiceImpl.java

```diff
@@ -82,26 +79,48 @@ public void initialize(String file) throws ServiceException, IOException {
   @Override
   public HashMap<String, String> getTags() {
     HashMap<String, String> tagMap = new HashMap<String, String>();
-    if (directory != null) {
-      for (Tag tag : directory.getTags()) {
-        tagMap.put(tag.getTagName(), tag.getDescription());
+    if (directories != null) {
+      for (Directory directory : directories) {
+        for (Tag tag : directory.getTags()) {
+          tagMap.put(tag.getTagName(), tag.getDescription());
```

Comment:
Not this PR fault, but I realize `tag.getDescription()` converts the values in the EXIF tag using some string formatting - see e.g. https://github.com/drewnoakes/metadata-extractor/blob/2.9.1/Source/com/drew/metadata/exif/ExifDescriptorBase.java#L262 which might prevent proper parsing in `JPEGReader`.